activerecord/lib/active_record/reflection.rb

```diff
@@ -298,6 +300,11 @@ def actual_source_reflection # FIXME: this is a horrible name
           self
         end
 
+        def klass_suppress_errors
```

Comment:
This doesn't feel great... can you elaborate on why it's required?

(Do we have a more general "candidate associations that might be my inverse" lookup elsewhere that we can lean on here?)